ecommerce/discounts_test.py

```diff
@@ -69,3 +110,94 @@ def test_discount_factory_adjustment(discounts, products):
                 discounted_price >= 0
                 and discounted_price == discount_logic.get_product_price(product)
             )
+
+
+def perform_discount_redemption(user, discount):
+    """Redeems a discount."""
+    order = Order(purchaser=user, total_price_paid=10)
+    order.save()
+
+    redemption = DiscountRedemption(
+        redeemed_discount=discount,
+        redemption_date=now_in_utc(),
+        redeemed_order=order,
+        redeemed_by=user,
+    )
+    redemption.save()
+
+
+def test_one_time_discount(user, onetime_discount):
+    """
+    Tests single-use discounts. These should be redeemable once, and then not
+    again.
+    """
+
+    assert onetime_discount.check_validity(user) == True
```

Comment:
Assertions against singletons like `True` and `False` should be done with `is`:
```suggestion
    assert onetime_discount.check_validity(user) is True
```

Same for similar usages in this file.